### PR TITLE
Implement voltage sensor

### PR DESCRIPTION
**Pull request by jordans6**
Small PR that adds battery sensor to sensor collection header and implementation files so that it can be used.

### CODE REVIEW

File: avionics/common/include/gpio.h [hunk 1]
```diff
@@ -34,7 +34,8 @@ enum class Pin {
     MAIN_CONTINUITY_READ = 14,
     DROGUE_IGNITOR = 35,
     DROGUE_CONTINUITY_TEST = 17,
-    DROGUE_CONTINUITY_READ = 34
+    DROGUE_CONTINUITY_READ = 34,
+    VOLTAGE_SENSOR = 36 // TODO: Check in Eagle for the pin the battery voltage divider is connected to
```
Comment (jesseli2002) [inline avionics/common/include/gpio.h:38]: maybe make a trello task for this if you haven't already?

File: avionics/common/src/gpio.cpp [hunk 2]
```diff
@@ -7,7 +7,7 @@
  *  drogue and main parachute releases, status LEDs, buzzer.
  *
  *  The drogue servo control function and main parachute release functions are
- *  also contained here. There is a function that implements continuity testing
+ *  also contained here. There is a function that im plements continuity testing
```
Comment (jesseli2002) [inline avionics/common/src/gpio.cpp:10]: :/

File: avionics/common/src/sensor_collection.cpp [hunk 3]
```diff
@@ -56,4 +58,9 @@ void SensorCollection::updateStatus() {
         raiseToStatus(status_, RocketStatus::NONCRITICAL_FAILURE);
         *status_bitfield_ |= 0x08;
     }
+    if (battery.getStatus() == SensorStatus::FAILURE) {
+        LOG_WARN("Voltage sensor failed");
+        raiseToStatus(status_, RocketStatus::NONCRITICAL_FAILURE);
+        *status_bitfield_ |= 0x04;
+    }
```
Comment (jesseli2002) [inline avionics/common/src/sensor_collection.cpp:66]: Not critical now, but maybe as a TODO - it'd make sense to measure the voltage before launch and if it's too low, raise a critical or non-critical error (e.g. we could have a voltage level below which we shouldn't launch, and one that we could launch but it'd be non-ideal)

**Review by jesseli2002 — COMMENTED** [review]
LGTM, but there's a few minor house-keeping things to do

File: avionics/common/include/gpio.h (hunk 1 above)
Comment (jordans6) [reply avionics/common/include/gpio.h:38]: https://trello.com/c/yJ85cmmG/374-check-that-voltage-sensor-pin-is-correct

File: avionics/common/src/sensor_collection.cpp (hunk 3 above)
Comment (jordans6) [reply avionics/common/src/sensor_collection.cpp:66]: TODO added and Trello created.

92fb4eb

https://trello.com/c/urowjTxS/375-check-voltage-level-before-launch

File: avionics/common/src/gpio.cpp (hunk 2 above)
Comment (jordans6) [reply avionics/common/src/gpio.cpp:10]: Don't know how that slipped by lol.  Fixed in latest commit

**Review by jesseli2002 — APPROVED** [review]

**Pull request merged**